src/nvim/option.c

```diff
@@ -1140,6 +1140,13 @@ static void do_set_option_string(int opt_idx, int opt_flags, char **argp, int ne
 
   // Get the new value for the option
   char *newval = stropt_get_newval(nextchar, opt_idx, argp, varp, origval, &op, flags);
+  // Validate winhl new value around by single quote
+  if (striequal(opt->shortname, "winhl")
```

Comment:
I don't think this is needed. when `parse_winhl_opt` returns false, the old value is already restored

Reply:
this can avoid screen redraw. keep highlight not changed.

Reply:
I don't think this block makes any sense. It only check for one specific case of invalid value.

Reply:
right. but how can i keep screen not redraw highlight ? even `parse_winhl_opt` return false the screen highlight changed.

Reply:
oh I know why redraw.. if `errmsg != NULL` and value restored. Is it must still run to `check_redraw_for` ? can return directly ? some option need this maybe ?

Reply:
seems that this optimization is not needed. removed it then.